src/core/server/http/http_service.mock.ts

```diff
@@ -32,7 +37,23 @@ import { OnPreResponseToolkit } from './lifecycle/on_pre_response';
 
 type BasePathMocked = jest.Mocked<InternalHttpServiceSetup['basePath']>;
 type AuthMocked = jest.Mocked<InternalHttpServiceSetup['auth']>;
-export type HttpServiceSetupMock = jest.Mocked<InternalHttpServiceSetup> & {
+
+export type HttpServiceSetupMock = jest.Mocked<
+  Omit<HttpServiceSetup, 'basePath' | 'createRouter'>
+> & {
+  basePath: BasePathMocked;
+  createRouter: jest.MockedFunction<() => RouterMock>;
+};
+export type InternalHttpServiceSetupMock = jest.Mocked<
+  Omit<InternalHttpServiceSetup, 'basePath' | 'createRouter'>
+> & {
+  basePath: BasePathMocked;
+  createRouter: jest.MockedFunction<(path: string) => RouterMock>;
+};
+export type HttpServiceStartMock = jest.Mocked<HttpServiceStart> & {
+  basePath: BasePathMocked;
+};
+export type InternalHttpServiceStartMock = jest.Mocked<InternalHttpServiceStart> & {
```

Comment:
These test types are more complex that I'd like them to be, however no combination of `jest.Mocked` and `DeeplyMockedKey` was reflecting the real mock type while respecting the 'concrete' one's signature.

I also modified the mocked `createRouter` signature to explicitly return a `RouterMock` (It was previously returning a plain `IRouter`, forcing consumers to forcecast to `jest.Mocked` in some tests). Downside is that `createSetupContact().createRouter.mockReturnValue(instance of IRouter)` will not be an invalid statement, but I didn't encounter any in the codebase, so I think it overall improves the testing experience.